lib/Frontend/Frontend.cpp

```diff
@@ -452,10 +470,10 @@ void CompilerInstance::setupOutputBackend() {
 }
 
 bool CompilerInstance::setup(const CompilerInvocation &Invoke,
-                             std::string &Error) {
+                             std::string &Error, ArrayRef<const char *> Args) {
```

Comment:
Can this be made work with `ParsedArgs`? 
Previously, downstream of  `parseArgs`, the rest of the compiler does not consult the raw argument values, for several good reasons (e.g. de-duplication). 
I think there's value to keeping that convention.

Reply:
Maybe ParsedArgs is better, but it has to be render backed into raw argument value for cache key currently. ParsedArg doesn't de-duplicate or handle ordering difference. The ideal solution is doing a canonicalization for CompilerInvocation

Reply:
Okay then, thanks, let's leave it as-is for now, but these details are adding up to make canonicalization increasingly important. 👍🏼